src/main/java/org/jenkinsci/plugins/fodupload/FodGlobalDescriptor.java

```diff
@@ -147,15 +148,17 @@ public FormValidation doTestApiKeyConnection(@QueryParameter(CLIENT_ID) final St
                                                  @QueryParameter(API_URL) final String apiUrl) {
         Jenkins.get().checkPermission(Jenkins.ADMINISTER);
         FodApiConnection testApi;
+        String plainTextClientId = Utils.retrieveSecretDecryptedValue(clientId);
+        String plainTextClientSecret= Utils.retrieveSecretDecryptedValue(clientSecret);
         if (Utils.isNullOrEmpty(baseUrl))
             return FormValidation.error("Fortify on Demand URL is empty!");
         if (Utils.isNullOrEmpty(apiUrl))
             return FormValidation.error("Fortify on Demand API URL is empty!");
-        if (Utils.isNullOrEmpty(clientId))

```

Comment:
Should client ID really be included here?

Reply:
No I don't think so I will remove along with the other non-sensitve fields.